src/databricks/labs/ucx/aws/access.py

```diff
@@ -354,3 +360,9 @@ def create_uber_principal(self, prompts: Prompts):
             logger.info(f"Cluster policy \"{cluster_policy.name}\" updated successfully")
         except PermissionError:
             self._aws_resources.delete_instance_profile(iam_role_name, iam_role_name)
+
+    def _generate_role_name(self, single_role: bool, role_name: str, location: str) -> str:
+        if single_role:
+            metastore_id = self._ws.metastores.current().as_dict()["metastore_id"]
+            return f"{role_name}_{metastore_id}"
+        return f"{role_name}_{location[5:]}"
```

Comment:
Does it make sense to always add the `metastore_id` and only the location if it has non-zero length?